packages/adapter-cloudflare/src/worker.js

```diff
@@ -17,11 +29,38 @@ export default {
 	 * @returns {Promise<Response>}
 	 */
 	async fetch(req, env, context) {
+		const options = /** @satisfies {Parameters<typeof server.respond>[1]} */ ({
+			platform: {
+				env,
+				context,
+				// @ts-expect-error webworker types from worktop are not compatible with Cloudflare Workers types
+				caches,
+				// @ts-expect-error the type is correct but ts is confused because platform.cf uses the type from index.ts while req.cf uses the type from index.d.ts
+				cf: req.cf
+			},
+			getClientAddress() {
+				return req.headers.get('cf-connecting-ip');
+			}
+		});
+
 		await server.init({
 			// @ts-expect-error env contains environment variables and bindings
-			env
+			env,
+			peers: ws?.peers,
+			publish: ws?.publish
 		});
 
+		if (req.headers.get('upgrade') === 'websocket' && ws) {
+			const hooks = await server.resolveWebSocketHooks(
+				req,
+				// @ts-ignore
+				options
+			);
+			resolve_websocket_hooks = () => hooks;
```

Comment:
Is there a possible race condition here? If two upgrade requests came in at the same moment, could the `resolve_websocket_hooks` called by the `ws` instance's `resolve` hook be (incorrectly) shared by the two requests?

It feels like if we had the ability to create the `ws` instance here instead of sharing it between all requests, we'd be able to eliminate this risk and also reduce some of the indirection. But IIUC we can only call it once otherwise peers can't talk to each other